A32NX/html_ui/Pages/A32NX_Core/A32NX_TransitionAltitude.js

```diff
@@ -0,0 +1,318 @@
+class A32NX_TransitionAltitude {
+    init() {
+        console.log('A32NX_TransitionAltitude init');
+        this.currentDeparture = "";
+        this.currentArrival = "";
+        this.offline = false;
+        this.serverStatus();
+    }
+
+    update(_deltaTime, _core) {
+        if (this.offline) {
+            this.offlineTransAlt();
+        }
+    }
+
+    serverStatus() {
+        if (!NXApi.hasTelexConnection()) {
+            this.offline = true;
+        }
+    }
+
+    offlineTransAlt() {
+        const Origin = NXDataStore.get("PLAN_ORIGIN", "");
+        const Destination = NXDataStore.get("PLAN_DESTINATION", "");
```

Comment:
Why are we storing this in localStorage? Why don't you just access it through the MCDU/FlightPlanManager when you need it again?

Reply:
I try to use that but that is not working.
in MCDU, can call with new A32NX_TransitionAltitude();
but in A32NX_TransitionAltitude.js, idk how to use that. like new flightPlanManger or mcdu.flightplanmanager etc... it is not working. 

so i just use local storage that already saved.
also i change some code so now when spawn on runway

Reply:
This needs to be accessible *somwhow*. We can't use localStorage for something like this that is already available. There is multiple issues around the data lifecycle with localStorage and we also have a hard maximum size we can't go over.

Reply:
Ok i try to do that. 
Before that, do you know how to call mcdu or flightplanmanager? I'm so annoying about that.

Reply:
if i change that other like flightplanmanger.getorigin etc... 
should i change telex body(it use that NXDataStore.get("PLAN_ORIGIN", "")?